### PR TITLE
DeferredSemaphore docstring documents constructor parameter

### PR DESCRIPTION
**Pull request by jisaacstone**
https://twistedmatrix.com/trac/ticket/9026#ticket

### CODE REVIEW

File: src/twisted/internet/defer.py [hunk 1]
```diff
@@ -1573,6 +1573,10 @@ class DeferredSemaphore(_ConcurrencyPrimitive):
     """
 
     def __init__(self, tokens):
+        """
+        @param tokens: initial value of C{tokens} and C{limit}
```
Comment (rodrigc) [inline src/twisted/internet/defer.py]: where is C{limit} set?

Comment (exarkun) [reply src/twisted/internet/defer.py]: Five lines below?  What do you mean?

Comment (markrwilliams) [reply src/twisted/internet/defer.py]: Maybe the confusion is that source code markup like `C{limit}` often refers to a callable's argument, but there's no `limit` argument.

Comment (markrwilliams) [reply src/twisted/internet/defer.py]: [Epytext can resolve `L{something}` in a method docstring to an instance variable `something`](http://epydoc.sourceforge.net/epytext.html) (See section 3.3).  While it's fine to use `C{...}` here, if you switch these to `L{tokens}` and `L{limit}`, pydoctor will generate links to those documented `@ivars`.  That would be pretty neat!

**Review by markrwilliams — COMMENTED** [review]
This is a good documentation improvement!

I left a suggestion on how to associate the `tokens` parameter with the two instance variables it sets.  Please also add a [topfile](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles).  I think this can be a `misc` change.

Thanks for the contribution!!

File: src/twisted/internet/defer.py (hunk 1 above)
Comment (jisaacstone) [reply src/twisted/internet/defer.py]: Did not know that, thanks!

**Comment by jisaacstone** [comment]
Topfile in :)

**Comment by markrwilliams** [comment]
@jisaacstone This looks good to me!  I've pushed this branch to twisted/twisted so the Buildbots will run it.  We'll see if that doesn't get codecov to change its mind about coverage.

**Pull request merged**